### DISCUSSION TITLE
Publish TopologyRefreshEvent to event bus

### DISCUSSION THREAD

**Discussion by rkeely** [redis/lettuce#2809] [Ideas]
I would like to know the reason of only publishing `TopologyRefreshEvent` to EventRecorder as implemented [here](
https://github.com/redis/lettuce/blob/8848d151ef902976faa8806e4dafd7d62478e9fd/src/main/java/io/lettuce/core/cluster/RedisClusterClient.java#L888). My app subscribe to all events on event bus but does not receive any `TopologyRefreshEvent` so I assume the current implementation of EventRecorder does not publish to event bus but only JFR, is that correct? Could we consider dual-publishing it to event bus as well? I am happy to create a PR if this idea sounds good.

Here are my reasons why we want this:
* People normally do not setup recording for JFR in production. We love out of box support without extra effort :D
* For Redis cluster with large number of nodes and with large number of clients, the toll on topology refresh will be heavy for both side. As application owner, we would like to monitor the occurrence of TopologyRefreshEvent and make sure there is no un-wanted  refresh. I know we can control the refresh periods but unexpected resonation from many clients could happen (e.g. when we scale out our client fleet). Currently we have to use the number connection events to infer the refresh. Being able to easily subscribe to the event could give an concrete answer.

**Comment by mp911de** [comment]
`EventBus` doesn't support currently long-lived events and therefore, event recording is limited to JFR currently. With a bit of redesign around `RecordableEvent`, it could be possible to hand in `RecordableEvent` into `EventBus` so that the event bus publishes the event upon `EventBus.publish`.

The design is to capture the start time of `TopologyRefreshEvent` and publish the event once the topology refresh has finished.

**Reply by rkeely** [reply]
Is this design already a work in progress, or you would be open to accepting a pull request for this? Happy to contribute.

**Reply by mp911de** [reply]
There's no work in progress. Feel free to submit a pull request and we can discuss over the actual changes.

**Reply by rkeely** [reply]
Created a commit for preview #2819 just to make sure we can align before any further changes.

Seems I can't add people as reviewer.

**Comment by mp911de** [comment]
I created an issue out of the discussion.
